assets/wizards/advertising/views/placements/index.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * Ads Global Placements Settings.
+ */
+
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { Component, Fragment } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import {
+	ActionCard,
+	Grid,
+	Notice,
+	SectionHeader,
+	SelectControl,
+	withWizardScreen,
+} from '../../../../components/src';
+
+/**
+ * Advertising management screen.
+ */
+class Placements extends Component {
```

Comment:
Nitpick since this is a new component: any particular reason to use a class-based component over a functional component?

Reply:
I might be behind on the latest React best practices because @adekbadek [asked me the same thing here](https://github.com/Automattic/newspack-plugin/pull/1208#discussion_r745454591) 😅 

I find that without a better state management structure that would provide HoCs or custom hooks, the stateful, or container, class-based components are better. Without these utilities I find functional components tend to get a bit messy with all the business logic inside.

Still, I can change that. Especially now that it has been brought up again haha

Reply:
2dc6bbf358dcc4ec82af89f1ead9adc5ae3c15c4 refactors the Placements component

Reply:
It's definitely somewhat subjective, and the React team [does not plan to deprecate or stop support class-based components anytime soon](https://reactjs.org/docs/hooks-faq.html#do-i-need-to-rewrite-all-my-class-components). But they are definitely moving slowly toward a hooks-based future:

> [In the longer term, we expect Hooks to be the primary way people write React components.](https://reactjs.org/docs/hooks-faq.html#should-i-use-hooks-classes-or-a-mix-of-both)

So while I agree that class-based components can sometimes be a bit more readable and easier to understand than hooks-based functional components, I think it's best that we move toward React's recommendation for _new_ components, at least. There's no real urgency to refactor existing class-based components unless we want to use hooks.

Reply:
Thank you for the links! To be honest, after I refactored it to a functional component I liked the result more than I expected 😅 .

This will be even better after we get #1272 going for every wizard 🎉